infra/relayer/src/purse.js

```diff
@@ -0,0 +1,696 @@
+
+    // If we have one in memory, go with that
+    if (txCount > 0) return txCount
+
+    // Check redis, if available
+    if (txCount === 0 && this.rclient && this.rclient.connected) {
+      const countFromRedis = await this.rclient.getAsync(
+        `${REDIS_TX_COUNT_PREFIX}${address}`
+      )
+      // null defense
+      if (countFromRedis) {
+        txCount = countFromRedis
+      }
+    } else {
+      logger.warn('Redis unavailable')
+    }
+
+    /**
+     * Finally, check with the JSON-RPC provider.  Chance of some pendings screwing this up if we
+     * get this far.  I'm at least sure that eth_pendingTransactions, is completely  not supported
+     * by Alchemy.  Doubtful from Infura as well.  And it's a poor method anyway with no arguments.
+     *
+     * EDGE CASE WARNING
+     * -----------------
+     *
+     * The only time this should be an issue is if the service is magically restarted, Redis is
+     * wiped or unavailable, all while there are transactions still in the pool waiting to be
+     * mined.
+     *
+     * Also, tiny chance of transaction replacement here if the gas price goes up during the above
+     * edge case.  That would be... like a dropped transaction in the client.  Perhaps not a deal
+     * breaker, but maybe some thought should be put into that in the future as volume goes up. The
+     * bigger the load, the likelier this becomes.
+     */
+    const w3txCount = parseInt(await this.web3.eth.getTransactionCount(address))
+    if (txCount < w3txCount) txCount = w3txCount
+
+    return txCount
+  }
+
+  /**
+   * Increment the known transaction count for an account
+   * @param address {string} - The account address to increment
+   */
+  async incrementTxCount(address) {
+    this._enforceExists(address)
+
+    this.accounts[address].txCount += 1
+    this.accounts[address].pendingCount += 1
+    if (this.rclient && this.rclient.connected) {
+      await this.rclient.incrAsync(`${REDIS_TX_COUNT_PREFIX}${address}`)
+    } else {
+      logger.warn('Redis unavailable')
+    }
+  }
+
+  /**
+   * Fund all the children from the master account
+   */
+  async fundChildren() {
+    if (this.children.length < 1) {
+      logger.warn('No children to fund')
+      return
+    }
+
+    const masterAddress = this.masterWallet.getChecksumAddressString()
+    const masterBalance = stringToBN(
+      await this.web3.eth.getBalance(masterAddress)
+    )
+
+    const maxFunding = BASE_FUND_VALUE.mul(numberToBN(this.children.length))
+
+    if (masterBalance.lt(maxFunding)) {
+      logger.warn(
+        `Master account ${masterAddress} does not have enough funding. Expected ${maxFunding}, has ${masterBalance}`
+      )
+      return
+    }
+
+    // If the equal split between all children is less than the base fund value
+    // use it instead of trying to overdraw
+    const eqSplit = masterBalance.div(numberToBN(this.children.length))
```

Comment:
Don't we also need to account for cost of gas to pay for the funding transactions to children ?

Reply:
Yes!  Thank you for catching that.